mono/utils/mono-path.c

```diff
@@ -177,27 +177,132 @@ mono_path_resolve_symlinks (const char *path)
 #endif
 }
 
+#ifndef HOST_WIN32
```

Comment:
Remove this hack and replace it by #ifdef HOST_WIN32

Reply:
This is a good technique to enable testing Win32 on Posix and vice versa.
The mainline code is as optimized as usual while a separate test variant enables more/easier testing.